dbms/src/Flash/Planner/Plans/PhysicalTableScan.cpp

```diff
@@ -28,6 +29,43 @@
 
 namespace DB
 {
+namespace
+{
+NamesWithAliases buildTableScanProjectionCols(
+    const NamesAndTypes & schema,
+    const NamesAndTypes & storage_schema)
+{
+    if unlikely (schema.size() != storage_schema.size())
+        throw TiFlashException(
+            fmt::format(
+                "The scan schema size ({}) from tidb is difference from the storage schema size ({}) from tiflash",
```

Comment:
Can we also show the table_id in the error message? I think it could be better for locating bugs for complicated SQLs

Reply:
ok, updated.